src/plots/geo/geo.js

```diff
@@ -136,21 +135,20 @@ proto.plot = function(geoData, fullLayout, promises) {
 };
 
 proto.onceTopojsonIsLoaded = function(geoData, geoLayout) {
-    var scattergeoData = [],
-        choroplethData = [];
+    var traceData = {};
 
     this.drawLayout(geoLayout);
 
     for(var i = 0; i < geoData.length; i++) {
         var trace = geoData[i];
-        var traceType = trace.type;
 
-        if(traceType === 'scattergeo') scattergeoData.push(trace);
-        else if(traceType === 'choropleth') choroplethData.push(trace);
+        traceData[trace.type] = traceData[trace.type] || [];
```

Comment:
:cow2: I usually write statements like these as:

```js
if(traceData[trace.type] === undefined) traceData[trace.type] = [];
```

not sure what's cleaner though. It would be nice to come to a consensus though.